src/charm.py

```diff
@@ -195,17 +217,23 @@ def remove_secret(self, scope, key) -> None:
         content[key] = Config.Secrets.SECRET_DELETED_LABEL
         secret.set_content(content)
 
+    def stop_mongos_service(self):
+        """Stop mongos service."""
+        container = self.unit.get_container(Config.CONTAINER_NAME)
+        container.stop(Config.SERVICE_NAME)
```

Comment:
This may raise a `ChangeError` - we should probably add a retry with `reraise=True` and catch on the caller side.

Reply:
See convo below about ChangeError on stop services in the comment thread below

In general though I am not a fan of Retrying if we fail to stop the charm on the caller side. I think if we fail to stop a service then something is wrong that is underlying and we should let it fail - I don't think retrying will fix it